thrift/shared.thrift

```diff
@@ -368,6 +368,7 @@ struct TaskListMetadata {
 struct WorkflowExecution {
   10: optional string workflowId
   20: optional string runId
+  30: optional string domain
```

Comment:
I might need a little more context on this, in the go library this works by scheduling a decision task response with the domain filled in: https://github.com/uber/cadence-idl/blob/master/proto/uber/cadence/api/v1/decision.proto#L143

But possibly I'm not understanding what you're trying to do